src/AutomationTests/OutputFileHelperUnitTests.cs

```diff
@@ -15,90 +15,98 @@ public class OutputFileHelperUnitTests
         [TestMethod]
         [Timeout(1000)]
         [ExpectedException(typeof(ArgumentNullException))]
-        public void OutputFileHelperCtor_NullSystemFactory_ThrowsException()
+        public void OutputFileHelperCtor_NullSystem_ThrowsException()
         {
-            new OutputFileHelper(outputDirectory: null, systemFactory: null);
+            new OutputFileHelper(outputDirectory: null, system: null);
         }
 
         [TestMethod]
         [Timeout(1000)]
         public void OutputFileHelperCtor_NullSystemDateTime_ThrowsException()
         {
-            var mockSystemFactory = new Mock<ISystemFactory>(MockBehavior.Strict);
-            mockSystemFactory.Setup(x => x.CreateSystemDateTime()).Returns<ISystemDateTime>(null);
+            var mockSystem = new Mock<ISystem>(MockBehavior.Strict);
+            mockSystem.Setup(x => x.DateTime).Returns<ISystemDateTime>(null);
 
-            Action action = () => new OutputFileHelper(outputDirectory: null, systemFactory: mockSystemFactory.Object);
+            Action action = () => new OutputFileHelper(outputDirectory: null, system: mockSystem.Object);
             var ex = Assert.ThrowsException<InvalidOperationException>(action);
             Assert.IsTrue(ex.Message.Contains("_dateTime"));
 
-            mockSystemFactory.VerifyAll();
+            mockSystem.VerifyAll();
         }
 
         [TestMethod]
         [Timeout(1000)]
         public void OutputFileHelperCtor_NullSystemEnvironment_ThrowsException()
         {
-            var mockSystemFactory = new Mock<ISystemFactory>(MockBehavior.Strict);
+            var mockSystem = new Mock<ISystem>(MockBehavior.Strict);
             var mockDateTime = new Mock<ISystemDateTime>(MockBehavior.Strict);
```

Comment:
>mockDateTime [](start = 16, length = 12)

A pattern that I like to use in unit tests: Add this to the class:

private static readonly ISystemDateTime InertSystemDateTime = new Mock<ISystemDateTime>(MockBehavior.String).Object;

Then in the unit tests that should never call this mock, pass it in:

mockSystem.Setup(x => x.DateTime).Returns(InertSystemDateTime);

That makes it explicit that you're using a mock that should never be invoked and shortens the unit test by 1 line for each inert mock that you use.

I noticed this because I was trying to determine if there should be a mockDateTime.VerifyAll at the bottom of the test. With the Inert* pattern, _every_ mock created in the test has a VerifyAll, so it's trivial to catch cases where a VerifyAll should exist.